src/Magnum/Platform/GlfwApplication.cpp

```diff
@@ -469,7 +469,8 @@ bool GlfwApplication::tryCreate(const Configuration& configuration, const GLConf
        If we are on Wayland, this is causing a segfault; a blinking window is
        acceptable in this case. */
     constexpr const char waylandString[] = "wayland";
-    if(std::strncmp(std::getenv("XDG_SESSION_TYPE"), waylandString, sizeof(waylandString)) != 0)
+    const char* xdgSessionType = std::getenv("XDG_SESSION_TYPE");
```

Comment:
```suggestion
    const char* const xdgSessionType = std::getenv("XDG_SESSION_TYPE");
```

:grin: ... other than that, thanks a lot!